lib/milestone-db/memory.js

```diff
@@ -0,0 +1,61 @@
+var MilestoneDB = require('./index');
+
+/**
+ * In-memory ShareDB milestone database
+ *
+ * Milestone snapshots exist to speed up Backend.fetchSnapshot by providing milestones
+ * on top of which fewer ops can be applied to reach a desired version of the document.
+ * This very concept relies on persistence, which means that an in-memory database like
+ * this is in no way appropriate for production use.
+ *
+ * The main purpose of this class is to provide a simple example of implementation,
+ * and for use in tests.
+ */
+module.exports = MemoryMilestoneDB;
+function MemoryMilestoneDB(options) {
+  MilestoneDB.call(this, options);
+
+  // Map form collection name -> doc id -> array of milestone snapshots
+  this._milestoneSnapshots = {};
+}
+
+MemoryMilestoneDB.prototype = Object.create(MilestoneDB.prototype);
+
+MemoryMilestoneDB.prototype.getMilestoneSnapshot = function (collection, id, version, callback) {
+  var milestoneSnapshots = this._getMilestoneSnapshotsSync(collection, id);
+
+  let milestoneSnapshot;
```

Comment:
`let` -> `var`, as ShareDB code doesn't use ES6 syntax yet - I don't think there's a specifically declared policy on browser compatibility.

Server-only code could certainly start using ES6 features, as the oldest supported version of Node is 6 LTS. It'd be nice to use Map for higher performance than object properties. However, it isn't always obvious what's server-only vs client-only vs shared.

Reply:
Ha whoops, yeah that was just a force of habit from my day-to-day work. I thought this would get picked up by the linter? (It certainly had a go at me for using the spread operator)